core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/model/MemStatementIterator.java

```diff
@@ -60,7 +60,7 @@
 	/**
 	 * The index of the last statement that has been returned.
 	 */
-	private volatile int statementIdx;
+	private int statementIdx;
```

Comment:
Intellij says that the use of a volatile variable here is not needed since the code that uses the variable is not synchronized. If this class is supposed to support multi-threaded access then the code that uses this variable also needs to be synchronized.